### PR TITLE
Add nested vertical list

### PR DESCRIPTION
**Pull request by mrstark14**

### CODE REVIEW

File: test/components/nested-vertical-list.html [hunk 1]
```diff
@@ -0,0 +1,935 @@
...
+    <div class="ui twenty-eighty-container faculty-vertical-list">
+        
+        <div class="first first-vertical-list">
+
+            <div class="vertical-list">
+                <div class="vertical-list-head ui sub-heading" id="1" onclick="head_click(this.id)">Biosketch</div>
+                <div class="vertical-list-nest">
+                    <div class="vertical-list-nest-content ui one-liner-captions" id="edudetails" onclick="content_click(this.id)">Educational Details</div>
+                    <div class="vertical-list-nest-content ui one-liner-captions" id="profbg" onclick="content_click(this.id)">Professional Background</div>
+                    <div class="vertical-list-nest-content ui one-liner-captions" id="adminpos" onclick="content_click(this.id)">Administrative Positions</div>
```
Comment (ParitoshKabra) [inline test/components/nested-vertical-list.html]: There is no need of a vertical-list-line class, just target the ```hr``` tag as a children of vertical-list

File: test/components/nested-vertical-list.html [hunk 2]
```diff
@@ -0,0 +1,935 @@
...
+<body class="light">
+    <div class="ui twenty-eighty-container faculty-vertical-list">
+        
+        <div class="first first-vertical-list">
+
+            <div class="vertical-list">
+                <div class="vertical-list-head ui sub-heading" id="1" onclick="head_click(this.id)">Biosketch</div>
+                <div class="vertical-list-nest">
+                    <div class="vertical-list-nest-content ui one-liner-captions" id="edudetails" onclick="content_click(this.id)">Educational Details</div>
+                    <div class="vertical-list-nest-content ui one-liner-captions" id="profbg" onclick="content_click(this.id)">Professional Background</div>
```
Comment (ParitoshKabra) [inline test/components/nested-vertical-list.html]: No need of `vertical-list-nest-content` as well, reasons are same as for hr tag

File: test/components/nested-vertical-list.html [hunk 3]
```diff
@@ -0,0 +1,935 @@
...
+	<title></title>
+</head>
+
+<body class="light">
+    <div class="ui twenty-eighty-container faculty-vertical-list">
+        
+        <div class="first first-vertical-list">
+
+            <div class="vertical-list">
+                <div class="vertical-list-head ui sub-heading" id="1" onclick="head_click(this.id)">Biosketch</div>
```
Comment (ParitoshKabra) [inline test/components/nested-vertical-list.html]: Same here

File: test/components/nested-vertical-list.html [hunk 4]
```diff
@@ -0,0 +1,935 @@
...
+                <div class="vertical-list-nest">
+                    <div class="vertical-list-nest-content ui one-liner-captions" id="edudetails" onclick="content_click(this.id)">Educational Details</div>
+                    <div class="vertical-list-nest-content ui one-liner-captions" id="profbg" onclick="content_click(this.id)">Professional Background</div>
+                    <div class="vertical-list-nest-content ui one-liner-captions" id="adminpos" onclick="content_click(this.id)">Administrative Positions</div>
+                </div>
+                <hr class="vertical-list-line" />
+                <div class="vertical-list-head ui sub-heading" id="2" onclick="head_click(this.id)">Research</div>
+                <div class="vertical-list-nest">
```
Comment (ParitoshKabra) [inline test/components/nested-vertical-list.html:28]: Let's discuss about this, do we really need this class as well? can't we target divs directly as `.vertical-list > div` ? @mrstark14 , also @ktg-123 @saurabhhere @ayu023ban  PTAL

**Comment by ParitoshKabra** [comment]
@mrstark14 refresh pe #wali cheej kaam nahi karti, woh fix karde

File: library/javascript/components/nested-vertical-list.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+const url = window.location.href.split('#')[0]
+
+const elements = document.getElementsByClassName('one-liner-captions')
```
Comment (saurabhhere) [inline library/javascript/components/nested-vertical-list.js]: specify the name of elements

File: library/javascript/components/nested-vertical-list.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+const elements1 = document.getElementsByClassName('list-sub-heading')
```
Comment (saurabhhere) [inline library/javascript/components/nested-vertical-list.js]: `element1`?

File: library/javascript/components/nested-vertical-list.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    const element = document.getElementById(id).nextSibling.nextSibling
+    const style = getComputedStyle(element);
+    if (style.display == 'none') {
+        element.style.display = 'block';
+    }
+    else if (style.display == 'block') {
+        element.style.display = 'none';
+    }
+    else {
+        console.log('invalid display style');
```
Comment (saurabhhere) [inline library/javascript/components/nested-vertical-list.js]: Remove console

File: library/css/components/faculty-profile.css [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  line-height: 1rem;
+  letter-spacing: 0.02rem;
+  margin-bottom: 1.125rem;
+  color: var(--faculty-card-info-color);
+  align-items: center;
+  display: grid;
+  grid-template-columns: 4% 96%;
+}
+
+.ui.ui.faculty-profile .ui.icon-typography div[onclick]:hover {
```
Comment (saurabhhere) [inline library/css/components/faculty-profile.css]: Multiple `ui`?

File: library/css/components/faculty-profile.css [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+.ui.faculty-profile > .twenty-eighty-container > .second > .description {
+  font-weight: 300;
+  font-size: 0.875rem;
+  line-height: 1.375rem;
+  color: var(--black);
+  margin-top: 1.5rem;
+  overflow-x: hidden;
+  max-height: 100px;
```
Comment (saurabhhere) [inline library/css/components/faculty-profile.css]: why in `px` use `rem`

File: library/css/components/nested-vertical-list.css [hunk 10]
```diff
@@ -0,0 +1,80 @@
+.vertical-list-nest {
+    display: none;
+}
+
+.vertical-list-head {
```
Comment (saurabhhere) [inline library/css/components/nested-vertical-list.css]: Try to use more selective css selectors using `>`

File: library/css/components/nested-vertical-list.css [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+.vertical-list-nest > div {
+    cursor: pointer;
+    margin-bottom: 21px;
+}
+
+.active-list-item {
+    color: #0067B3DE !important;
+}
+
+.vertical-list {
```
Comment (saurabhhere) [inline library/css/components/nested-vertical-list.css]: More selective css selector, same name can be used by someone else also in other components

File: library/javascript/components/nested-vertical-list.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    }
+}
+
+function content_click(id) {
+    const element1 = document.getElementsByClassName('active-list-item')
+    if (element1.length != 0) {
```
Comment (saurabhhere) [inline library/javascript/components/nested-vertical-list.js]: I guess we can use `!==` here?

File: library/css/components/nested-vertical-list.css [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+@media only screen and (max-width: 630px) {
+
+    .faculty-vertical-list .first {
+        width: 85% !important;
+        padding: 5vw 6vw 5vw 6vw !important;
+    }
+
+    .vertical-list {
```
Comment (saurabhhere) [inline library/css/components/nested-vertical-list.css]: More selective in the whole css file

**Review by saurabhhere — CHANGES_REQUESTED** [review]

**Comment by saurabhhere** [comment]
@mrstark14 Around `1300px` pr page pr horizontal scroll aa rha hai, I guess second container ki wajah se aa raha, vo bhi dekh lena

**Pull request merged**